moveit_commander/src/moveit_commander/move_group.py

```diff
@@ -339,16 +323,14 @@ def set_pose_target(self, pose, end_effector_link=""):
             if type(pose) is PoseStamped:
                 old = self.get_pose_reference_frame()
                 self.set_pose_reference_frame(pose.header.frame_id)
-                ok = self._g.set_pose_target(
-                    conversions.pose_to_list(pose.pose), end_effector_link
-                )
+                ok = self._g.set_pose_target(pose.pose, end_effector_link)
                 self.set_pose_reference_frame(old)
             elif type(pose) is Pose:
+                ok = self._g.set_pose_target(pose, end_effector_link)
+            else:
                 ok = self._g.set_pose_target(
-                    conversions.pose_to_list(pose), end_effector_link
+                    conversions.list_to_pose(pose), end_effector_link
```

Comment:
Before, it was possible to specify a pose as a list? I think we should deprecate such an API.